nextcord/http.py

```diff
@@ -268,152 +663,250 @@ async def request(
-                    kwargs["data"] = form_data
-
-                try:
-                    async with self.__session.request(method, url, **kwargs) as response:
-                        _log.debug(
-                            "%s %s with %s has returned %s",
-                            method,
-                            url,
-                            kwargs.get("data"),
-                            response.status,
-                        )
-
-                        # even errors have text involved in them so this is safe to call
-                        data = await json_or_text(response)
-
-                        # check if we have rate limit header information
-                        is_global = bool(response.headers.get("X-RateLimit-Global", False))
-
-                        limit = response.headers.get("X-RateLimit-Limit", "")
-                        remaining = response.headers.get("X-Ratelimit-Remaining", "")
-
-                        bucket = response.headers.get("X-RateLimit-Bucket")
-                        if remaining == "0" and response.status != 429:
-                            # we've depleted our current bucket
-                            delta = utils.parse_ratelimit_header(response, use_clock=self.use_clock)
+        auth = headers.get("Authorization")
+
+        # If a global rate limit for this authorization doesn't exist yet, make it.
+        if (global_rate_limit := self._global_rate_limits.get(auth)) is None:
+            global_rate_limit = self._make_global_rate_limit(auth, self._default_max_per_second)
+
+        global_rate_limit = cast(GlobalRateLimit, global_rate_limit)
+
+        # If a rate limit for this url path doesn't exist yet, make it.
+        if (url_rate_limit := self._get_url_rate_limit(route.method, route, auth)) is None:
+            url_rate_limit = self._make_url_rate_limit(route.method, route, auth)
+
+        max_retry_count = 5
+        rate_limit_path = (
+            route.method,
+            route.bucket,
+            _get_logging_auth(auth),
+        )  # Only use this for logging.
+        ret: Any | None = None
+        response: aiohttp.ClientResponse | None = None
+
+        # The loop is to allow migration to a different RateLimit if needed.
+        # If we hit this loop max_retry_count times, something is wrong. Either we're migrating buckets way
+        #  too much, 429s keep getting hit, or something is internally wrong.
+        for retry_count in range(max_retry_count):  # To prevent infinite loops.
+            should_retry = False
+            try:
+                async with global_rate_limit:
+                    async with url_rate_limit:
+                        # This check is for asyncio.gather()'d requests where the rate limit can change.
+                        if (
+                            temp := self._get_url_rate_limit(route.method, route, auth)
+                        ) is not url_rate_limit and not None:
```

Comment:
```suggestion
                        ) not in (url_rate_limit, None):
```

Reply:
Not too sure about that one, identity checks are much faster so are better when you can.

Reply:
*SIGH*, why do I occasionally do this. Fixed.